am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/defaultrolesetup/GrantDefaultPermissionIntegrationTest.java

```diff
@@ -0,0 +1,79 @@
+package integration.uk.gov.hmcts.reform.amlib.defaultrolesetup;
+
+import integration.uk.gov.hmcts.reform.amlib.base.IntegrationBaseTest;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.amlib.enums.AccessManagementType;
+import uk.gov.hmcts.reform.amlib.enums.RoleType;
+import uk.gov.hmcts.reform.amlib.enums.SecurityClassification;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermissionGrant;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.createDefaultPermissionGrant;
+import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.createReadPermissionsForAttribute;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
+class GrantDefaultPermissionIntegrationTest extends IntegrationBaseTest {
+
+    @Test
+    void whenAddResourceDefinitionIsCalledAddNewEntryIntoDatabase() {
+        defaultRoleService.addRole(
+            ROLE_NAME, RoleType.RESOURCE, SecurityClassification.Public, AccessManagementType.ROLEBASED);
+
+        defaultRoleService.grantDefaultPermission(createDefaultPermissionGrant());
+
+        assertThat(countDefaultPermissions()).isEqualTo(1);
```

Comment:
I am afraid that checking number of rows is not sufficient. It does not give us assurance that parameters query are bound correctly.